CHANGELOG.md

```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+General:
+  * Binary builds only when `GHC >= 8.10` due to easier `haskeline >= 0.8 && < 0.9`.
```

Comment:
```suggestion
  * The executable is only buildable with `GHC >= 8.10` due to its dependency `haskeline >= 0.8 && < 0.9`.
```

Reply:
Well, there is a rule in Cabal that makes the build to compile build only `if` GHC >= 8.10, but `when` has more proper connotation.

"buildable with `GHC >= 8.10`" would mean that build of executable fails when `GHC < 8.10`, which is not true, from the maintainers and users perspective executable just not gets build and person may not even notice it. So that is why it is "builds only when `GHC >= 8.10`".

---

"The executable is only buildable with" means that it is impossible, when in reality, it may be supported, it was just chosen as easier to not worry about it.

"Easier", means technically if someone needs it - they can participate to support old `haskeline` in parallel, literally https://github.com/haskell-nix/hnix/pull/540#issuecomment-634977813 "it seems that disabling the executable might just be good enough" means that it is easier to drop the executable builds on older GHC versions.

Reply:
I agree that I must switch "binary" word for "executable", since it is more proper term:

```fish
file /nix/store/<hash>/bin/file
/nix/store/<hash>/bin/file: ELF 64-bit LSB executable
...
```

Reply:
I previously purposefully avoided the "executable" since I do not like the connotation of the word.

I switched the "binary" for "executable" in PR and would use "executable" further, until dictionary of terms would change.